File: org.boblycat.frida.core/src/java/org/boblycat/frida/core/BinaryCodeChunk.java
package org.boblycat.frida.core;

/**
 * Created: Mar 19, 2010
 * <p/>
 * Part of Frida IRE.
 * Copyright (c) 2010, Karl Trygve Kalleberg, Ole André Vadla Ravnås
 * Licensed under the GNU General Public License, v3
 *
 * @author: karltk@boblycat.org
 */
public class BinaryCodeChunk {

    private final long baseAddress;
    private final byte[] rawData;

    public BinaryCodeChunk(long baseAddress, byte[] rawData) {
        this.baseAddress = baseAddress;
        this.rawData = rawData;
    }

    public byte byteAt(int index) { return rawData[index]; }
    public byte[] getBytes() { return rawData; }
    public long getBaseAddress() { return baseAddress; }
    public int length() { return rawData.length; }
}

File: org.boblycat.frida.core/src/scala/org/boblycat/frida/core/BinaryCodeChunk.scala
package org.boblycat.frida.core

/**
 * Created: Mar 19, 2010
 * <p/>
 * Part of Frida IRE.
 * Copyright (c) 2010, Karl Trygve Kalleberg, Ole André Vadla Ravnås
 * Licensed under the GNU General Public License, v3
 *
 * @author: karltk @boblycat.org
 */

@serializable
class BinaryCodeChunk(val baseAddress : Long, val rawData : Array[Byte]) {

  def getBaseAddress = baseAddress

  def byteAt(index : Int): Byte = {
    return rawData(index)
  }

  def getBytes: Array[Byte] = rawData
  def length: Int = rawData.length
}